index.Rmd

```diff
@@ -131,58 +124,57 @@ In its most basic use-case, `read_csv` expects that the data file:
 > This has several benefits that we'll discuss in further detail later in the course.
 
-## Assigning value to a data frame
+## Giving a data frame a name
 
-When we loaded the language data collected in the 2016 Canadian census in R above using `read_csv`, we did not give this data frame a name, so it was 
-just printed to the screen and we cannot do anything else with it. That isn't very useful; what we would like to do is give a name to the data frame that `read_csv` outputs
-so that we can use it later for analysis and visualization. 
+When we loaded the language data collected in the 2016 Canadian census in R above using `read_csv`, we did not give this data frame a name. Therefore the data was just printed on the screen, and we cannot do anything else with it. That isn't very useful; we would like to give a name to the data frame that `read_csv` outputs to use later for analysis and visualization. 
 
-To assign name to something in R, there are two possible ways---using either the assignment symbol (`<-`) or the equals symbol (`=`). From a style perspective, 
-the assignment symbol is preferred and is what we will use in this course. When we name something in R using the assignment symbol, `<-`, we do not need to surround 
-it with quotes like the filename. This is because we are formally telling R about this word and giving it a value. Only characters and words that act as values need 
-to be surrounded by quotes.
+There are two possible ways to assign a name to something in R ---using either the assignment symbol (`<-`) or the equals symbol (`=`). From a style perspective, the assignment symbol is preferred and is what we will use in this course. When we name something in R utilizing the assignment symbol, `<-`, we do not need to surround it with quotes like the file name because we are formally telling R about this word and giving it a value. Only characters and words that act as values need to be surrounded by quotes.
 
 Let's now use the assignment symbol to give the name `can_lang` to the language data collected in the 2016 Canadian census data frame that we get from `read_csv`.
 
 ```{r load_data_with_name, message=FALSE}
 can_lang <- read_csv("data/can_lang.csv")
 ```
 
-Wait a minute! Nothing happened this time! Or at least it looks like that. But actually, something did happen: the data was read in and now has the name `can_lang` associated with it. 
-And we can use that name to access the data frame and do things with it. First we will type the name of the data frame to print it to the screen.
+Wait a minute! Nothing happened this time! Or at least it looks like that. But actually, something did happen: the data was read in and now has the name `can_lang` associated with it. And we can use that name to access the data frame and do things with it. We can type the name of the data frame to print the first few rows on the screen.
 
 
 ```{r print}
 can_lang
 ```
 
+
+
 ## Creating subsets of data frames with `select` & `filter`
```

Comment:
Can we do something interesting here with `select` and `filter`? Perhaps something that will lead us to create our final visualization in this chapter?

Perhaps we can state our question here, which is "what ten Aboriginal languages were most often reported in 2016 as mother tongues in Canada and how many people speak each of them?" and then try to answer them first by creating a table with `filter` & `select`? And then we can bring up the idea that although tables are great, visualizations are often even more insightful. And then we make the visualization?

So if we did that, then the story would go first `filter` to get the Aboriginal languages, and then `select` just `language` and `mother_tongue`? Then we do `slice` to get the top 10? I suggest `slice` instead of `slice_max` as we use `slice` a lot in later worksheets for the course, and it is a bit more general. We have to use `reorder` for the plot, and then we might get questions, why doesn't `slice_max` change the bar order? Which is a good question, but I don't think we want them to be asking that yet.

>Note: another thing that I think we do want to add in here is a richer description introducing the question we are asking. Perhaps something along the lines of Aboriginals were in Canada before colonization, and there were many groups with X different languages. Sadly, colonization has led to the loss of many of these. In the 2016 census, Y Aboriginal languages were recorded as still being spoken. To learn more see this resource (this Canadian/National Geographic reference looks really good: https://www.canadiangeographic.ca/article/mapping-indigenous-languages-canada and maybe another?).

Reply:
@leem44 - Now wondering if we want to start off the chapter with our question? Specifically, just before we teach them to load data. Then we have a question we want to answer, some data in a file that can answer it. Now we just need to use R to load the data and do some wrangling and viz to answer it!

Reply:
I really like this idea! I will make changes